src/components/Search/Search.tsx

```diff
@@ -1,20 +1,43 @@
-import React, { useEffect, useState } from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
 import './Search.css';
 
-export const Search = () => {
-  const [search, setSearch] = useState((localStorage.getItem('search') as string) || 'Http cats');
+type TSearch = {
+  onCards: Dispatch<SetStateAction<never[]>>;
+  error: Dispatch<SetStateAction<null>>;
+};
+
+export const Search = (props: TSearch) => {
+  const [search, setSearch] = useState((localStorage.getItem('search') as string) || '');
+  const { onCards } = props;
+  const [input, setInput] = useState('');
 
   const onValueChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
     setSearch(e.target.value);
   };
 
-  useEffect(() => {
+  const handleSubmit = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    setInput(search);
     localStorage.setItem('search', search);
-  }, [search]);
+  };
+
+  useEffect(() => {
+    fetch(`https://rickandmortyapi.com/api/character/?name=${search}`).then(async (res) => {

```

Comment:
If you are using .then keep this way all code, don't mix with async. For e.g. code with .then:
fetch(url)
 .then(res => {
   if (!res.ok) throw Error(...)
   res.json())
   }
 .then(data => setData(data))
 .catch(err => setError(err.message))

Reply:
fix it